portal-kernel/src/com/liferay/portal/kernel/upgrade/DuplicateRemovalUpgradeProcess.java

```diff
@@ -0,0 +1,260 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.kernel.upgrade;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Jorge Avalos
+ */
+public class DuplicateRemovalUpgradeProcess extends UpgradeProcess {
+
+	public DuplicateRemovalUpgradeProcess(String tableName, String columns) {
+		this(tableName, columns, null);
+	}
+
+	public DuplicateRemovalUpgradeProcess(
+		String tableName, String columns, String orderByClause) {
+
+		_tableName = tableName;
+		_columns = columns;
+		_orderByClause = orderByClause;
+	}
+
+	@Override
+	protected void doUpgrade() {
+		_removeDuplicates();
+	}
+
+	protected List<Map<String, String>> getDuplicatesSQL(
+		String[] indexDuplicateColumns) {
+
+		List<Map<String, String>> queryResult = new ArrayList<>();
+
+		String[] columns = _columns.split(", ");
+
+		StringBundler sb = new StringBundler();
+
+		sb.append("select * from ");
+		sb.append(_tableName);
+		sb.append(" where ");
+
+		for (int i = 0; i < indexDuplicateColumns.length; i++) {
+			sb.append(columns[i]);
+
+			if (indexDuplicateColumns[i] == null) {
+				sb.append(" IS NULL ");
+			}
+			else {
+				sb.append(" = '");
+				sb.append(_escape(indexDuplicateColumns[i]));
```

Comment:
@javalosjr96 why the escape is needed? Is it because we don't know they type?

Reply:
@achaparro 
The escape would be for values that contain DB strings(check the array).

When I was testing with different tables, the query would fail because of a character in a string.